cs/testing-with-travis.texy

```diff
@@ -0,0 +1,192 @@
+Průběžné testování s Nette Tester
+********************************
+
+/--div .[perex]
+Znáte to. Napsali jste testy, které teď procházejí, poplácali se po zádech a váš spánek začal být mnohem klidnější. Máte jistotu, že váš kód spolehlivě funguje. A pak se to stane. Hodiny ukazují 3:15 a do nočního ticha se zakousne nepříjemný zvuk drnčení mobilu na nočním stolku. Volá šéf. Zase to nefunguje. Zalije vás studený pot. Vždyť na to máme testy! V tu chvíli se vynoří vzpomínka na páteční odpoledne - spěcháte domů a zapomínáte spustit testy po poslední úpravě.
+
+Abyste předešli podobným zážitkům, ukážeme si, jak průběžně testovat pomocí populárního nástroje [Travis CI](https://travis-ci.org/):
+
+- nastavení testovacího prostředí
+- instalace závislostí pomocí Composeru
+- spouštění testů pomocí Nette/Tester
+- nastavení dalších služeb
+- [TL;DR](#Výsledný soubor)
+\--
+
+Jak funguje Travis CI
+=====================
+
+Travis CI (dále jen Travis) je velmi populární nástroj pro automatizaci [průběžné integrace](http://cs.wikipedia.org/wiki/Průběžná_integrace). Je plně integrován do prostředí GitHubu.
+
+Travis se nastavuje pomocí konfiguračního souboru `.travis.yml` umístěného v kořenovém adresáři projektu. Obsahuje několik bloků, které si podrobněji popíšeme.
+
+Bloky jsou spouštěny v pořadí `before_install`, `install`, `before_script`, `script`, `after_success` nebo `after_failure` a nakonec `after_script`. Povinný je pouze blok `script`.
+
+K výsledku testu můžete v nastavení přistoupit pomocí proměnné `$TRAVIS_TEST_RESULT`.
+
+Pro podrobnější informace si přečtěte [oficiální dokumentaci |http://docs.travis-ci.com].
+
+Nastavení jazyka
+================
+
+Nejprve je potřeba pomocí `language` Travis informovat, pro jaký jazyk má prostředí nastavit. Na dalších řádcích uvádíme, pro jaké verze PHP budeme chtít náš projekt otestovat. Testy pak budou puštěny pro každou verzi zvlášť. Protože Nette Tester podporuje i [HHVM](http://hhvm.com), můžeme ho přidat také.
+
+/--
+language: php
+
+php:
+  - 5.3
+  - 5.4
+  - 5.5
+  - 5.6
+  - hhvm
+\--
+
+Proměnné prostředí
+==================
+
+V sekci `env` nastavujeme proměnné prostředí. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. My budeme používat proměnnou `TESTER_PHP_BIN`, jejíž hodnotu poté získáme pomocí prefixu `$`. 
+
+
+
+/--
+env:
+  - TESTER_PHP_BIN=php
+  - TESTER_PHP_BIN=php-cgi
+\--
+
+Instalace závislostí
+====================
+
+.[caution]
+Návod předpokládá, že závislosti vašeho projektu jsou spravovány pomocí nástroje Composer a v kořenovém adresáři obsahuje validní `composer.json`. Zároveň musíte mít v závislostech uvedený nette/tester. Jak nastavit Composer najdete v kapitole [#Composer].
```

Comment:
Část o vyžadování `--dev` je hned pod tím, kde vysvětluji, s jakými options composer pouštím. Je to nedostatečné?